esp-hal-embassy/src/time_driver.rs

```diff
@@ -156,10 +156,11 @@ impl Driver for EmbassyTimer {
     }
 
     fn set_alarm(&self, alarm: AlarmHandle, timestamp: u64) -> bool {
-        // we sometimes get called with `u64::MAX` for apparently not yet initialized
-        // timers which would fail later on
+        // If `embassy-executor/integrated-timers` is enabled and there are no pending
+        // timers, embassy still calls `set_alarm` with `u64::MAX`. By returning
+        // `true` we signal that no re-polling is necessary.
         if timestamp == u64::MAX {
-            return false;
+            return true;
```

Comment:
interesting ... docs could be a bit better at explaining the meaning of the result 🤔

Reply:
the reason it calls `set_timer(u64::MAX)` is to cancel an existing timer from a previous poll if any. This can happen if on a previous poll a task was waiting on a timer, and then on this poll it's not anymore. This can happen if you cancel the timer, for example if it was before waiting on either a gpio event or a timer, and now is waiting on a gpio event only.

Not calling `set_timer(u64::MAX)` would do one "useless" poll when the stale timer expires.

Reply:
So this piece of code would ideally unset the alarm, not just "do nothing, return true".

Reply:
I've posted https://github.com/embassy-rs/embassy/pull/3255 which clarifies this, please let me know what you think.

Reply:
It's an interesting assumption that we have any active alarms, but the explanation makes sense and I guess it points at the current PR being incomplete. Thank you.